src/Data/Newtype.purs

```diff
@@ -33,51 +34,36 @@ import Data.Semigroup.Last (Last(..))
 -- | unwrap <<< wrap = id
 -- | wrap <<< unwrap = id
 -- | ```
-class Newtype t a | t -> a where
-  wrap :: a -> t
-  unwrap :: t -> a
+class Newtype :: Type -> Type -> Constraint
+class (Coercible t a, Coercible a t) <= Newtype t a | t -> a
```

Comment:
One of those Coercible constraints will become redundant when (if?) we merge https://github.com/purescript/purescript/pull/3930.

Reply:
Right, but until that's done, I'm leaving this in.